src/ddoc_cache/src/ddoc_cache_lru.erl

```diff
@@ -241,80 +189,87 @@ lru_start(Key, DoInsert) ->
             ddoc_cache_entry:recover(Key)
     end.
 
-trim(_, 0) ->
-    full;
-trim(St, MaxSize) ->
+trim(#st{} = St, 0) ->
+    {full, St};
+trim(#st{} = St, MaxSize) ->
     CurSize = ets:info(?CACHE, memory) * erlang:system_info(wordsize),
     if
         CurSize =< MaxSize ->
-            ok;
+            {ok, St};
         true ->
             case ets:first(?LRU) of
                 {_Ts, Key, Pid} ->
-                    remove_entry(St, Key, Pid),
-                    trim(St, MaxSize);
+                    St1 = remove_entry({Key, Pid}, St),
+                    trim(St1, MaxSize);
                 '$end_of_table' ->
-                    full
+                    {full, St}
             end
     end.
 
-remove_entry(St, Key, Pid) ->
-    #st{
-        pids = Pids
-    } = St,
+get_entries(DbName, #{} = Dbs) ->
+    case Dbs of
+        #{DbName := DDocIds} ->
+            Fun = fun(_, Keys, Acc1) -> maps:to_list(Keys) ++ Acc1 end,
+            maps:fold(Fun, [], DDocIds);
+        _ ->
+            []
+    end.
+
+do_refresh(#{} = DDocIdsMap, [_ | _] = DDocIdList) ->
+     Fun = fun(DDocId) ->
+         case DDocIdsMap of
+             #{DDocId := Keys} ->
+                 maps:foreach(fun(_, Pid) -> ddoc_cache_entry:refresh(Pid) end, Keys);
+             _ ->
+                 ok
+         end
+     end,
+     lists:foreach(Fun, [no_ddocid | DDocIdList]).
+
+remove_entry({Key, Pid}, #st{pids = Pids, dbs = Dbs} = St) ->
```

Comment:
Note: `remove_entry/2` args were changed so that it can be directly used in a lists:foldl callback.